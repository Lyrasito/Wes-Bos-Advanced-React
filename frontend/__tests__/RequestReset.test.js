import { mount } from "enzyme";
import wait from "waait";
import RequestResetComponent, {
  REQUEST_RESET_MUTATION,
} from "../components/RequestReset";
import toJSON from "enzyme-to-json";
import { MockedProvider } from "react-apollo/test-utils";

const mocks = [
  {
    request: {
      query: REQUEST_RESET_MUTATION,
      variables: { email: "marie@email.com" },
    },
    result: {
      data: {
        requestReset: {
          message: "success",
          __typename: "Message",
        },
      },
    },
  },
];

describe("<RequestResetComponent />", () => {
  it("renders and matches snapshot", () => {
    const wrapper = mount(
      <MockedProvider>
        <RequestResetComponent />
      </MockedProvider>
    );
    const form = wrapper.find("form[data-test='form']");
    expect(toJSON(form)).toMatchSnapshot();
  });
  it("calls the mutation", async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <RequestResetComponent />
      </MockedProvider>
    );
    wrapper.find("input").simulate("change", {
      target: { name: "email", value: "marie@email.com" },
    });
    wrapper.find("form").simulate("submit");
    await wait();
    wrapper.update();
    expect(toJSON(wrapper.find("p"))).toMatchSnapshot();
  });
});
