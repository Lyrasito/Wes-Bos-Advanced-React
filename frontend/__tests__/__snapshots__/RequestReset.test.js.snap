// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestResetComponent /> calls the mutation 1`] = `
<p>
  Request sent successfully! Please check your email for a reset link.
</p>
`;

exports[`<RequestResetComponent /> renders and matches snapshot 1`] = `
<form
  className="Form-sc-1xszr8q-0 krRxky"
  data-test="form"
  onSubmit={[Function]}
>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <h2>
      Request a password reset
    </h2>
    <DisplayError
      error={Object {}}
    />
    <label
      htmlFor="email"
    >
      Email
       
      <input
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="email"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Send Request
    </button>
  </fieldset>
</form>
`;
